packages/uicore/src/components/PillToggle/PillToggle.tsx

```diff
@@ -20,10 +27,56 @@ export interface PillToggleProps<T> {
   onChange: (val: T) => void
   disableToggle?: boolean
   className?: string
+  disableToggleReasonIcon?: IconName
+  showDisableToggleReason?: boolean
+  disableToggleReasonContent?: React.ReactElement
 }
 
 export const PillToggle = <T,>(props: PillToggleProps<T>): React.ReactElement => {
-  const { selectedView, onChange, disableToggle = false, className = '', options } = props
+  const {
+    selectedView,
+    onChange,
+    disableToggle = false,
+    className = '',
+    options,
+    disableToggleReasonIcon = 'error-outline',
+    showDisableToggleReason = false,
+    disableToggleReasonContent
+  } = props
+
+  const renderInvalidBadge = (pillToggleValue: T) => {
+    if (disableToggle && showDisableToggleReason && selectedView !== pillToggleValue) {
+      return (
+        <Popover interactionKind={PopoverInteractionKind.HOVER} position={Position.BOTTOM} className={Classes.DARK}>
+          <Icon
+            name={disableToggleReasonIcon}
+            size={12}
+            className={css.disableToggleReasonIcon}
+            data-testid="invalid-icon"
+            intent={Intent.DANGER}
+          />
+          {disableToggleReasonContent ?? (
+            <Container padding="medium">
+              <Layout.Vertical width={325} padding={{ left: 'small' }}>
+                <Text
+                  width={284}
+                  color={Color.GREY_0}
+                  margin={{ bottom: 'small' }}
+                  font={{ size: 'normal', weight: 'light' }}>
+                  The Visual Editor is disabled because of the errors in the YAML file.
+                </Text>
+                <Text width={284} color={Color.GREY_0} font={{ size: 'normal', weight: 'light' }}>
+                  Fix all the errors indicated in the YAML Editor to enable the Visual mode.
+                </Text>
```

Comment:
To support internationalisation, these strings should be passed down as prop.

Reply:
The idea is to give some default UI here. Consumers should pass their own UI for tooltip ideally and that will handle our concern for strings.